src/main/java/com/google/api/codegen/discovery/Document.java

```diff
@@ -133,6 +147,22 @@ public Schema dereferenceSchema(Schema schema) {
   }
 
   /** @return the auth instructions URL. */
+  @JsonIgnore @Nullable private Node parent;
+
+  public Node parent() {
+    return parent;
+  }
+
+  // Package private for use by other Node objects.
+  void setParent(Node parent) {
+    this.parent = parent;
+  }
+
+  /** @return whether or not to version the module. */
+  @JsonProperty("id")
+  public abstract String id();
```

Comment:
Would sort this and the other functions alphabetically to match the current style.